### ISSUE TITLE
file sync not triggered after redeploy

### ISSUE THREAD

**Issue by jlewi** [GoogleContainerTools/skaffold#3454]
I believe I observed the following behavior.

1. Configured Skaffold to trigger file sync 

    * Here's my [skaffold.yaml](https://github.com/jlewi/code-intelligence/blob/multi_model/Label_Microservice/skaffold.yaml)

1. I started skaffold in dev mode

1. I modified my code
1. Skaffold triggered a file copy for the modified files
1. I modified my deployment 
1. Skaffold triggered an update of the deployment
    * However skaffold didn't retrigger the file sync so the files in my deployment were out of date. 

I believe I observed the above but I didn't try to specifically reproduce it and verify it.  

### Expected behavior

After an update to the deployment files should be resync'd so the filesystem in the deployment contains any changes since the image was built

### Actual behavior

File sync not triggered automatically after rebuild.

(I could work around this my modifying the file to retrigger)

### Information

- Skaffold version: v1.1.0
- Operating system: ...
- Contents of skaffold.yaml:

```yaml
# Reference: https://skaffold.dev/docs/references/yaml/
apiVersion: skaffold/v2alpha1
kind: Config
metadata:
  name: label-microservice
build:
  artifacts:
  - image: gcr.io/issue-label-bot-dev/bot-worker
    # Set the context to the root directory. 
    # All paths in the Dockerfile should be relative to this one.
    context: ..
    # Automatically sync python files to the container. This should avoid
    # the need to rebuild and redeploy when the files change.
    sync:
        infer:        
        - '**/*.py'
    kaniko:
      dockerfile: Label_Microservice/deployment/Dockerfile.worker
      buildContext:
        gcsBucket: issue-label-bot-dev_skaffold-kaniko
      env: 
        - name: GOOGLE_APPLICATION_CREDENTIALS
          value: /secret/user-gcp-sa.json
      cache: {}
  cluster:
    # pullSecret can be set to a local file from which the pull secret should be created.
    pullSecretName: user-gcp-sa
    # TODO(jlewi): This should be changed for each developer; or maybe we should create a reusable one?
    namespace: jlewi-dev
    resources:
      requests:
        cpu: 8
        memory: 16Gi

deploy:
  kustomize:
    path: deployment/overlays/dev
```

### Steps to reproduce the behavior

See above

1. a clonable repository with the sample skaffold project
https://github.com/jlewi/code-intelligence/blob/multi_model/Label_Microservice/skaffold.yaml

**Comment by feloy** [comment]
Here is a simple config and steps to reproduce the bug: https://github.com/feloy/skaffold-sync-bug

```
skaffold dev
[...]
Watching for changes...
[app-54488bd6b6-dwnrp app] 1
[app-54488bd6b6-dwnrp app] 1
                                                # echo 2 > content.txt
Syncing 1 files for gcr.io/your-project/app:[...]
Watching for changes...
[app-54488bd6b6-dwnrp app] 2
[app-54488bd6b6-dwnrp app] 2
                                                # sed -i 's/poll/poll2/' app.yaml 
                                                # or kubectl delete pod -l app=app
Starting deploy...
 - deployment.apps/app configured
Watching for changes...
[app-7f5db9d7d4-w5m7z app] 1   # /!\ We get the image built at startup, without synced content.txt file
[app-7f5db9d7d4-w5m7z app] 1

```

**Comment by briandealwis** [comment]
@feloy I tried your example with Skaffold 1.7 and HEAD and it all works:
```
[app-5958869cc6-66rvp app] 1
[app-5958869cc6-66rvp app] 1
Syncing 1 files for gcr.io/your-project/app:3e3860902e6af31c47e9e62106d21afbf794b02bdf7477ced6603f0afaba3cef
Watching for changes...
[app-5958869cc6-66rvp app] 2
[app-5958869cc6-66rvp app] 2
Syncing 1 files for gcr.io/your-project/app:3e3860902e6af31c47e9e62106d21afbf794b02bdf7477ced6603f0afaba3cef
Watching for changes...
[app-5958869cc6-66rvp app] 3
[app-5958869cc6-66rvp app] 3
Syncing 1 files for gcr.io/your-project/app:3e3860902e6af31c47e9e62106d21afbf794b02bdf7477ced6603f0afaba3cef
Watching for changes...
[app-5958869cc6-66rvp app] 4
[app-5958869cc6-66rvp app] 4
Syncing 1 files for gcr.io/your-project/app:3e3860902e6af31c47e9e62106d21afbf794b02bdf7477ced6603f0afaba3cef
Watching for changes...
[app-5958869cc6-66rvp app] 5
```

Would you mind trying a newer Skaffold and see if you can still reproduce?  If it's still present, please run with `-v debug` to output more verbose logs and attach here.

**Comment by briandealwis** [comment]
Is it possible you had clock skew between your node and your local machine?

**Comment by briandealwis** [comment]
Closing.  Will re-open if new information comes to light.

One thing to be aware of: there was a problem with Skaffold (with Go actually) around mixed-case filepaths in case-insensitive filesystems, such as typically seen on Darwin (#4108).  This was fixed in Skaffold 1.10.1.
